extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -64,9 +59,23 @@ class AmpYoutube extends AMP.BaseElement {
         this.element.getAttribute('video-id')),
         'The data-videoid attribute is required for <amp-youtube> %s',
         this.element);
+  }
 
+  /** @override */
+  prerenderCallback() {
     if (!this.getPlaceholder()) {
-      this.buildImagePlaceholder_();
+      // Image load promise.
+      // Cancel this if the layoutCallback promise finished before it did.
+      return this.buildImagePlaceholder_();
```

Comment:
This is an example I was talking about. In theory this is somewhat of a regression from the previous code.